src/hotspot/share/gc/shenandoah/c1/shenandoahBarrierSetC1.cpp

```diff
@@ -221,26 +221,21 @@ void ShenandoahBarrierSetC1::load_at_resolved(LIRAccess& access, LIR_Opr result)
   }
 
   // 3: apply keep-alive barrier if ShenandoahSATBBarrier is set
-  if (ShenandoahSATBBarrier) {
-    bool is_weak = (decorators & ON_WEAK_OOP_REF) != 0;
-    bool is_phantom = (decorators & ON_PHANTOM_OOP_REF) != 0;
+  if (ShenandoahSATBBarrier && ShenandoahBarrierSet::need_keep_alive_barrier(decorators, type)) {
```

Comment:
I wonder if it is actually valid to check for SATB barrier here. `ShenandoahBarrierSetC2::load_at_resolved` only does `need_keep_alive_barrier` check.